sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/UploadBlobResult.cs

```diff
@@ -8,14 +8,20 @@ namespace Azure.Containers.ContainerRegistry.Specialized
     /// </summary>
     public class UploadBlobResult
     {
-        internal UploadBlobResult(string digest)
+        internal UploadBlobResult(string digest, long size)
         {
             Digest = digest;
+            Size = size;
         }
 
         /// <summary>
         /// The digest of the uploaded blob, calculated by the registry.
         /// </summary>
         public string Digest { get; }
+
+        /// <summary>
+        /// The size of the uploaded blob.
+        /// </summary>
+        public long Size { get; }
```

Comment:
I am not very familiar with ACR and had a question on chunk sizing. I am a little confused how the size of the uploaded blob could differ from MaxChunkSize passed in the UploadBlobOptions. Are we letting the user know what size chunk they uploaded (but wouldn't they know already)? If MaxChunkSize is not given, the size of the uploaded blob would be 4 Mb. If MaxChunkSize is passed in as 0, I think nothing will be uploaded and there wouldn't be a result?

Reply:
Also curious about this - it doesn't look like this is returned from the service so we are essentially just returning the stream.Length to the user. Not sure if this is valuable information to include here.

Reply:
> so we are essentially just returning the stream.Length to the user

Yes, that's fair.  My thinking was that this is useful to have during an image push operation, to author the image manifest.  It's true that the caller might already have the size of their stream.  If for some reason they're passing through a non-seekable stream, they might not be able to call Length on it?

Reply:
I don't think we support non-seekable streams 😄

Reply:
>  I am a little confused how the size of the uploaded blob could differ from MaxChunkSize passed in the UploadBlobOptions. 

The chunk size represents fragments of the blob and MaxChunkSize lets callers say how big those fragments should be so they can tune the upload operation, e.g. based on characteristics of their system.  The size of the uploaded blob represents the sum of all the chunk sizes for the entire upload operation.